core/create/src/prompts/oidc.ts

```diff
@@ -0,0 +1,274 @@
+        type: 'confirm',
+        initial: true,
+        message:
+          "Is it alright if we try to query AWS for IAM permissions from the previous IAM deployment user for this project? If none are found we'll just use some default permissions."
+      },
+      { onCancel }
+    )
+    if (cancelled) {
+      return true
+    }
+    if (fetchOldPermissions) {
+      const previousDocument = await getPreviousIAMPermissions(
+        vaultEnv[`AWS_ACCESS_KEY_ID_${awsAccount}`],
+        vaultEnv[`AWS_SECRET_KEY_${awsAccount}`],
+        serviceName
+      )
+      if (previousDocument) {
+        winstonLogger.info(
+          'Found the permissions assigned to an IAM user for CI deployments previously. Reusing these permissions for your OIDC role policy.'
+        )
+        cloudformationTemplate.setIn(
+          ['Resources', 'CircleciOIDCRolePolicy', 'Properties', 'PolicyDocument'],
+          previousDocument
+        )
+      } else {
+        winstonLogger.info(
+          "Couldn't find any old IAM permissions to reuse for this project. Falling back to the default permissions recommended by Cloud Enablement."
+        )
+      }
+    }
+  }
+
+  let circleciProjectId
+  try {
+    circleciProjectId = await getCircleCiProjectId(vaultEnv.CIRCLECI_AUTH_TOKEN)
+  } catch (err) {
+    winstonLogger.error(`Failed to automatically determine the ID for your CircleCI prject: ${err}`)
+    circleciProjectId = (
+      await prompt(
+        {
+          name: 'circleId',
+          type: 'text',
+          message:
+            "You can instead enter the project ID manually. You can find the ID by going to your project's settings in the CircleCI UI and copying the ID displayed at Overview > Project ID."
+        },
+        { onCancel }
+      )
+    ).circleId
+    if (cancelled) {
+      return true
+    }
+  }
+
+  const templatePath = `cloudformation/${teamName}/${stackTags.systemCode}-deployment-role.yaml`
+  const configPath = `config/composer/custom/${teamName}/global/${stackTags.systemCode}-deployment-role.yaml`
+
+  const composerConfig = {
+    template_path: templatePath,
+    parameters: {
+      SystemCode: stackTags.systemCode,
+      CircleciProjectId: circleciProjectId,
+      CircleciOrgName: 'Financial-Times'
+    },
+    stack_tags: stackTags
+  }
+
+  const upstreamRepo = `aws-composer-account-${awsAccount}`
+  const changes = new Map([
+    [templatePath, { mode: FILE_MODE, content: cloudformationTemplate.toString() }],
+    [configPath, { mode: FILE_MODE, content: YAML.stringify(composerConfig) }]
+  ])
+  winstonLogger.info(
+    `Will send the following changes to the ${styles.URL(
+      `https://github.com/Financial-Times/${upstreamRepo}`
+    )} repository: ${changes}`
+  )
+  const { prConfirm } = await prompt(
+    { name: 'prConfirm', type: 'confirm', message: 'Is that okay?' },
+    { onCancel }
+  )
+  if (prConfirm) {
+    const octokit = new Octokit({ auth: vaultEnv.GITHUB_ACCESS_TOKEN })
```

Comment:
**question:** Where is the access token from?

Reply:
I created a fine-grained GitHub access token in `next-team`'s account. It only allows you to create git commits and PRs in the two `aws-composer` repositories we're supporting.